diff --git a/drivers/video/drm/sunxi_drm_hdmi.c b/drivers/video/drm/sunxi_drm_hdmi.c
index f30f9de..ac0d393 100644
--- a/drivers/video/drm/sunxi_drm_hdmi.c
+++ b/drivers/video/drm/sunxi_drm_hdmi.c
@@ -223,6 +223,11 @@ static int _sunxi_drv_hdmi_clock_on(struct sunxi_drm_hdmi *hdmi)
 		clk_prepare_enable(pclk->rst_bus_sub);
 	}
 
+	if (!IS_ERR_OR_NULL(pclk->clk_hdmi_bus)) {
+		hdmi_trace("hdmi drv bus clock enable\n");
+		clk_prepare_enable(pclk->clk_hdmi_bus);
+	}
+
 	if (!IS_ERR_OR_NULL(pclk->clk_hdmi_24M)) {
 		hdmi_trace("hdmi drv clock 24M enable\n");
 		clk_prepare_enable(pclk->clk_hdmi_24M);
